src/main/java/de/dennisguse/opentracks/viewmodels/IntervalStatistics.java

```diff
@@ -0,0 +1,80 @@
+package de.dennisguse.opentracks.viewmodels;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import de.dennisguse.opentracks.content.data.TrackPoint;
+import de.dennisguse.opentracks.util.LocationUtils;
+import de.dennisguse.opentracks.util.UnitConversions;
+
+public class IntervalStatistics {
+    List<Interval> intervalList;
+
+    public void build(List<TrackPoint> trackPointList, float distanceInterval) {
+        intervalList = new ArrayList<>();
+
+        if (trackPointList == null || trackPointList.size() == 0) {
+            return;
+        }
+
+        Interval interval = new Interval();
+        for (int i = 1; i < trackPointList.size(); i++) {
+            TrackPoint prevTrackPoint = trackPointList.get(i - 1);
+            TrackPoint trackPoint = trackPointList.get(i);
+
+            if (LocationUtils.isValidLocation(trackPoint.getLocation()) && LocationUtils.isValidLocation(prevTrackPoint.getLocation())) {
+                interval.distance += prevTrackPoint.distanceTo(trackPoint);
+                interval.time += trackPoint.getTime() - prevTrackPoint.getTime();
+
+                if (interval.distance >= distanceInterval) {
+                    float adjustFactor = distanceInterval / interval.distance;
+                    Interval adjustedInterval = new Interval(interval);
+                    adjustedInterval.distance *= adjustFactor;
```

Comment:
For Interval:
public void adjust(float adjustFactor) {
  distance *= adjustFactor;
  time *= adjustFactor;
}